apps/condo/domains/organization/components/DeleteEmployeeButtonWithReassignmentModel.tsx

```diff
@@ -0,0 +1,248 @@
+    softDeleteAction: IUseSoftDeleteActionType<OrganizationEmployee>
+    disabled?: boolean
+    employee: Pick<OrganizationEmployee, 'name'> & { organization?: 
+    Pick<OrganizationEmployee['organization'], 'id'>, user?: 
+    Pick<OrganizationEmployee['organization'], 'id'> }
+    activeTicketsOrganizationEmployeeCount: number
+}
+
+const ERROR_NOTIFICATION_TYPE = 'error'
+const WARNING_NOTIFICATION_TYPE = 'warning'
+const SUCCESS_NOTIFICATION_TYPE = 'success'
+
+const waitBetweenRequsted = async () => await sleep(1000)
+
+export const DeleteEmployeeButtonWithReassignmentModel: React.FC<IDeleteEmployeeButtonWithReassignmentModel> = ({
+    buttonContent,
+    softDeleteAction,
+    disabled = false,
+    employee,
+    activeTicketsOrganizationEmployeeCount,
+}) => {
+    const intl = useIntl()
+    const ConfirmReassignEmployeeTitle = intl.formatMessage({ id: 'employee.reassignTickets.title' })
+    const ConfirmDeleteButtonLabel = intl.formatMessage({ id: 'employee.reassignTickets.buttonContent.deleteUser' })
+    const ConfirmReassignTicketsButtonLabel = intl.formatMessage({ id: 'employee.reassignTickets.buttonContent.reassignTickets' })
+    const SearchPlaceholderLabel = intl.formatMessage({ id: 'EmployeesName' })
+    const AlertTitleLabel = intl.formatMessage({ id: 'employee.reassignTickets.alert.title' }, { employeeName: employee?.name || null })
+    const CountShortLabel = intl.formatMessage({ id: 'global.count.pieces.short' })
+    const AlertMessageLabel = intl.formatMessage({ id: 'employee.reassignTickets.alert.message' })
+    const NotificationTitleWarningLabel = intl.formatMessage({ id: 'employee.reassignTickets.notification.title.warning' })
+    const NotificationTitleErrorLabel = intl.formatMessage({ id: 'employee.reassignTickets.notification.title.error' })
+    const NotificationTitleSuccessLabel = intl.formatMessage({ id: 'employee.reassignTickets.notification.title.success' })
+    const NotificationMessageWarningLabel = intl.formatMessage({ id: 'employee.reassignTickets.notification.message.warning' })
+    const NotificationMessageErrorLabel = intl.formatMessage({ id: 'employee.reassignTickets.notification.message.error' })
+    const NotificationMessageSuccessLabel = intl.formatMessage({ id: 'employee.reassignTickets.notification.message.success' })
+
+    const [notificationApi, contextHolder] = notification.useNotification()
+
+    const employeeUserId = employee?.user?.id || null
+    const employeeOrganizationId = employee?.organization?.id || null
+
+    const getTicketReassignData = (ticket: Ticket) => {
+        const resultObj = {}
+        if (ticket?.executor?.id === employeeUserId) resultObj['executor'] = { connect: { id: newEmployeeUserId } }
+        if (ticket?.assignee?.id === employeeUserId) resultObj['assignee'] = { connect: { id: newEmployeeUserId } }
+        return resultObj
+    }
+
+    const getNotificationInfo = useCallback((notificationType: string, updatedTicketsCount = null) => {
```

Comment:
```suggestion
    const getNotificationInfo = useCallback((notificationType: 'error' | 'warning' | 'success', updatedTicketsCount = null) => {
```
It would be better to specify specific values in the type

Reply:
and then you won't need to put these values into constants.
(but if you want, you can leave them)